src/pool.c

```diff
@@ -703,47 +704,54 @@ uint64_t
 account_wc(const char *address)
 {
     account_t *account = NULL;
+    uint64_t wc = 0;
     pthread_rwlock_rdlock(&rwlock_acc);
     HASH_FIND_STR(accounts, address, account);
-    pthread_rwlock_unlock(&rwlock_acc);
     if (!account)
-        return 0;
-    return (uint64_t)account->worker_count;
+        goto bail;
+    wc = (uint64_t)account->worker_count;
+bail:
+    pthread_rwlock_unlock(&rwlock_acc);
+    return wc;
 }
 
 void
 account_rl(char *rig_list_out, const char *address)
 {
-   char rig_list[MAX_RIG_ID*MAX_RIG_ID+128] = "";
-   uint32_t rig_count = 0;
-   account_t *account = NULL;
-   pthread_rwlock_rdlock(&rwlock_acc);
-   HASH_FIND_STR(accounts, address, account);
-   pthread_rwlock_unlock(&rwlock_acc);
-   if (!account)
-     goto bail;
-   client_t *c = (client_t*)gbag_first(bag_clients);
-   while ((c = gbag_next(bag_clients, 0)))
-   {
-      if (strstr(c->address, address) != NULL)
-      {
-	 if (rig_count < MAX_RIG_ID && c->rig_id != 0 )
-	 {
-             if(rig_count > 0)
-                strcat(rig_list, ",\"");
-	     else
-                strcat(rig_list, "\"");
-             strncat(rig_list, c->rig_id, MAX_RIG_ID);
-	     strcat(rig_list, "\"");
-	     rig_count++;
-	 }
-      }
-   }
+    char *body = rig_list_out;
+    char *end = rig_list_out + (MAX_RIG_ID*MAX_RIGS+(MAX_RIGS*3));
+    uint32_t rig_count = 0;
+    account_t *account = NULL;
+
+    pthread_rwlock_rdlock(&rwlock_acc);
+    if (strlen(address) > ADDRESS_MAX)
+        goto bail;
+    HASH_FIND_STR(accounts, address, account);
+    if (!account)
+        goto bail;
+
+    client_t *c = (client_t*)gbag_first(bag_clients);
+    while ((c = gbag_next(bag_clients, 0)))
```

Comment:
And this loop would continue whilst c is not NULL and there is still room in the buffer.

Reply:
Sorry 3 should be 4 (to account for the NULL).